### PR TITLE
new(scap): Support gVisor no_events mode/full proc scan for nodriver

### PR DESCRIPTION
**Pull request by gnosek**
**What type of PR is this?**

> Uncomment one (or more) `/kind <>` lines:

> /kind bug

> /kind cleanup

> /kind design

> /kind documentation

> /kind failing-test

/kind feature

**Any specific area of the project related to this PR?**

> Uncomment one (or more) `/area <>` lines:

> /area API-version

> /area build

> /area CI

> /area driver-kmod

> /area driver-bpf

> /area driver-modern-bpf

> /area libscap-engine-bpf

/area libscap-engine-gvisor

> /area libscap-engine-kmod

> /area libscap-engine-modern-bpf

/area libscap-engine-nodriver

> /area libscap-engine-noop

> /area libscap-engine-source-plugin

> /area libscap-engine-savefile

> /area libscap-engine-udig

> /area libscap

> /area libpman

> /area libsinsp

> /area tests

> /area proposals

**Does this PR require a change in the driver versions?**

> /version driver-API-version-major

> /version driver-API-version-minor

> /version driver-API-version-patch

> /version driver-SCHEMA-version-major

> /version driver-SCHEMA-version-minor

> /version driver-SCHEMA-version-patch

**What this PR does / why we need it**:

This PR is a smaller patch to achieve the same thing as https://github.com/falcosecurity/libs/pull/986. Instead of introducing a no_events mode for all engines, it introduces one just for gVisor. kmod/bpf/modern_bpf can use the nodriver engine instead (with the new full_proc_scan=true flag) to achieve the same result.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Review by Andreagit97 — APPROVED** [review]
Thank you for this!
/approve

**Comment by Andreagit97** [comment]
cc @LucaGuerra 
/hold

File: userspace/libscap/scap-int.h [hunk 1]
```diff
@@ -56,6 +56,7 @@ struct scap
 	scap_userlist* m_userlist;
 	struct ppm_proclist_info* m_driver_procinfo;
 	uint32_t m_fd_lookup_limit;
+	bool m_minimal_scan;
```
Comment (LucaGuerra) [inline userspace/libscap/scap-int.h:59]: I think I'm missing this field initialization to false (except for the `!engine_params || !engine_params->full_proc_scan` case)

Comment (LucaGuerra) [reply userspace/libscap/scap-int.h:59]: Or perhaps this is `calloc`'d or already zero initialized in the common part

Comment (deepskyblue86) [reply userspace/libscap/scap-int.h:59]: https://github.com/falcosecurity/libs/blob/master/userspace/libscap/scap.c#L557 there

Comment (LucaGuerra) [reply userspace/libscap/scap-int.h:59]: Answering myself, we use at least C99 in the scap c part, and the `stdbool.h` header is included by scap. Works for me, I think we can merge it.

**Pull request merged**